client/electron/go_vpn_tunnel.ts

```diff
@@ -103,27 +106,39 @@ export class GoVpnTunnel implements VpnTunnel {
     if (checkProxyConnectivity) {
       this.isUdpEnabled = await checkUDPConnectivity(
         this.transportConfig,
+        this.gatewayAdapterIp,
+        this.gatewayAdapterIndex,
         this.isDebugMode
       );
     }
     console.log(`UDP support: ${this.isUdpEnabled}`);
 
     console.log('starting routing daemon');
-    await Promise.all([
-      this.tun2socks.start(this.transportConfig, this.isUdpEnabled),
-      this.routing.start(),
-    ]);
+    const gateway = await this.routing.start();
+    this.gatewayAdapterIp = gateway?.gatewayIp;
+    this.gatewayAdapterIndex = gateway?.gatewayIndex;
+    await this.startTun2socks();
   }
 
-  networkChanged(status: TunnelStatus) {
+  networkChanged(
+    status: TunnelStatus,
+    gatewayIp?: string,
```

Comment:
See my comment on whether we actually need to have both here.

Reply:
Removed